utils/self-extracting-executable/decompressor.cpp

```diff
@@ -387,22 +382,14 @@ int main(int argc, char* argv[])
     }
     else
     {
-        /// move decompressed file instead of this binary and apply command
-        char bash[] = "sh";
-        char executable[] = "-c";
-
-        /// length of forwarded args
-        size_t length = 0;
-        for (int i = 1; i < argc; ++i)
-            length += strlen(argv[i]);
-
-        /// mv filename.decompressed filename && filename agrs...
-        char command[3 + strlen(argv[0]) + 14 + strlen(argv[0]) + 4 + strlen(argv[0]) + length + argc];
-        fillCommand(command, argc, argv, length);
-
-        /// replace file and call executable
-        char * newargv[] = { bash, executable, command, nullptr };
-        execvp(bash, newargv);
+        const char * const cmd_fmt = "mv %s %s.delete && cp %s.decompressed %s && rm %s.delete %s.decompressed";
```

Comment:
This does not looks safe.
- What if someone have a file with this name already? Yes, it is unlikely, but doing so automatically not good.
- Lack of escaping.

Why not write it in C/C++? `mv` is simple, `rm` too, `cp` is not that simple but not complex either (P.S. you may use `copyfile` on linux and a write/read loop on other OSes)

P.S. have you considered to add some C++ here, to make code cleaner?

Reply:
As I remember the initial requirement was to make decompressor mostly C-ish, but since than it quite diverged from this goal... a little more diversion probably will not be that much of a deal...

Reply:
regarding overwriting files - compressed can contain many different files and will already overwrite files with same names - I don't see special treatment for temporaries is required in this regard

Reply:
>As I remember the initial requirement was to make decompressor mostly C-ish, but since than it quite diverged from this goal... a little more diversion probably will not be that much of a deal...

I think that if it will be just C++ w/o dbms library, than it should be completely OK, since it will allow to simplify the code.

>regarding overwriting files - compressed can contain many different files and will already overwrite files with same names - I don't see special treatment for temporaries is required in this regard...

Indeed, but it wasn't great either (from my point of view)

> though I will fix it...

Great, thanks!